### PR TITLE
p3 sin extras todavía, pondré extra si me logro despertar a tiempo

### PR DESCRIPTION
**Pull request by mikelsegura**

### CODE REVIEW

File: MikelSeguraElizalde/p3/pos_tagging.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+from sklearn_crfsuite import CRF
+from random import randrange
+from sklearn.metrics import classification_report
+
+corpus_file = open("corpus_otomi", "r")
+raw_corpus = corpus_file.read()
+raw_sentences = (raw_corpus.replace(' ','')).split("\n")
+sentences_words_unsplitted = [sentence.split("],[[") for sentence in raw_sentences]
+sentences_words_splitted = [[word.split("],") for word in sentence] for sentence in sentences_words_unsplitted]
+sentences_words_splitted.pop() #necesario para quitar un vacío al final
```
Comment (umoqnier) [inline MikelSeguraElizalde/p3/pos_tagging.py:16]: Una forma muy peculiar de cargar el corpus de texto a objetos de python. Creo que está algo propensa a errores, pero funciona de momento :0

Recomiendo usar algo como `eval()`

File: MikelSeguraElizalde/p3/pos_tagging.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    features = {
+        'word': word,
+        'word[:3]': word[:3], # tratando de detectar sufijos
+        'word[:2]': word[:2],
+        'word[-3:]': word[-3:], # tratando de detectar prefijos
+        'len(word)': len(word), # importante al ser el otomí muy sintético
+        'word.lower()': word.lower(),
+        'word.isupper()': word.isupper(),
+        'word.istitle()': word.istitle(),
+        'word.isdigit()': word.isdigit(),
```
Comment (umoqnier) [inline MikelSeguraElizalde/p3/pos_tagging.py:44]: Creo que estas features no son muy útiles en vista de que el corpus no contiene números y todo está en minúsculas

File: MikelSeguraElizalde/p3/pos_tagging.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+y_test_flat = [label for sent_labels in y_test for label in sent_labels]
+y_pred_flat = [label for sent_labels in y_pred for label in sent_labels]
+
+# Evaluate the model
+report = classification_report(y_true=y_test_flat, y_pred=y_pred_flat)
+print(report)
+
+# Mostrar ejemplos
+stop_examples = ''
+while not stop_examples:
```
Comment (umoqnier) [inline MikelSeguraElizalde/p3/pos_tagging.py:89]: Este bucle es infinito. No hay forma de salir más que con CTRL+C

File: MikelSeguraElizalde/p3/corpus_otomi
Comment (umoqnier) [inline MikelSeguraElizalde/p3/corpus_otomi:1]: No era necesario subir el corpus :1234:

**Review by umoqnier — APPROVED** [review]
Funciona y tiene un *accuracy* aceptable. Aunque la usabilidad del programa puede mejorar :p

Calificación: 10
